lib/__tests__/watch.test.ts

```diff
@@ -0,0 +1,191 @@
+      commandOptions: {},
+      filePaths: [],
+    };
+
+    watch(accountId, src, dest, options);
+
+    expect(chokidar.watch).toHaveBeenCalledWith(src, {
+      ignoreInitial: true,
+      ignored: expect.any(Function),
+    });
+  });
+
+  it('should trigger folder upload on initialization', async () => {
+    const accountId = 123;
+    const src = 'src-folder';
+    const dest = 'dest-folder';
+    const options = {
+      mode: MODE.draft,
+      remove: false,
+      disableInitial: false,
+      notify: '',
+      commandOptions: {},
+      filePaths: [],
+    };
+    const postInitialUploadCallback = jest.fn();
+
+    (uploadFolder as jest.Mock).mockResolvedValueOnce([]);
+
+    await watch(accountId, src, dest, options, postInitialUploadCallback);
+
+    expect(uploadFolder).toHaveBeenCalledWith(
+      accountId,
+      src,
+      dest,
+      {},
+      options.commandOptions,
+      options.filePaths,
+      options.mode
+    );
+    expect(postInitialUploadCallback).toHaveBeenCalled();
+  });
+
+  it('should log ready when watcher is ready', () => {
+    const accountId = 123;
+    const src = 'src-folder';
+    const dest = 'dest-folder';
+    const options = {
+      mode: MODE.draft,
+      remove: false,
+      disableInitial: true,
+      notify: '',
+      commandOptions: {},
+      filePaths: [],
+    };
+
+    watch(accountId, src, dest, options);
+
+    expect(chokidarMock.on).toHaveBeenCalledWith('ready', expect.any(Function));
+    chokidarMock.on.mock.calls[0][1]();
+    expect(logger.log).toHaveBeenCalledWith(expect.stringContaining('ready'));
+  });
+
+  it('should upload file when file is added', () => {
+    const accountId = 123;
+    const src = 'src-folder';
+    const dest = 'dest-folder';
+    const options = {
+      mode: MODE.draft,
+      remove: false,
+      disableInitial: true,
+      notify: '',
+      commandOptions: {},
+      filePaths: [],
+    };
+
+    watch(accountId, src, dest, options);
+
+    expect(chokidarMock.on).toHaveBeenCalledWith('add', expect.any(Function));
+    const addCallback = chokidarMock.on.mock.calls[1][1];
+    const filePath = '/some-file-path.html';
+
+    addCallback(filePath);
+
+    expect(logger.debug).toHaveBeenCalledWith(
+      expect.stringContaining('Attempting to upload')
+    );
+  });
+
+  it('should handle file change event and upload file', () => {
+    const accountId = 123;
+    const src = 'src-folder';
+    const dest = 'dest-folder';
+    const options = {
+      mode: MODE.draft,
+      remove: false,
+      disableInitial: true,
+      notify: '',
+      commandOptions: {},
+      filePaths: [],
+    };
+
+    watch(accountId, src, dest, options);
+
+    expect(chokidarMock.on).toHaveBeenCalledWith(
+      'change',
+      expect.any(Function)
+    );
+    const changeCallback = chokidarMock.on.mock.calls[2][1];
+    const filePath = 'changed-file-path.html';
+
+    changeCallback(filePath);
+
+    expect(logger.debug).toHaveBeenCalledWith(
```

Comment:
Curious to hear what others think about this too, but just leaving the same comment about us not needing to test the log output.

Reply:
Addressed in https://github.com/HubSpot/hubspot-local-dev-lib/pull/195/commits/a54efdd0207005059c9f34db23cfaa33e8dc5de5